src/theory/quantifiers/quantifiers_rewriter.cpp

```diff
@@ -1211,6 +1222,23 @@ Node QuantifiersRewriter::getVarElimEqBv(Node lit,
     // solve for the variable on this path using the inverter
     std::vector<unsigned> path;
     Node slit = binv.getPathToPv(lit, cvar, path);
+    // check if the path had a kind that does not preserve equivalence of the
+    // overall literal
+    if (!disallowedKinds.empty())
+    {
+      Node curr = lit;
+      for (size_t i = 0, npath = path.size(); i < npath; i++)
+      {
+        Trace("quant-velim-bv") << "On path: " << curr << std::endl;
+        if (disallowedKinds.find(curr.getKind()) != disallowedKinds.end())
+        {
+          slit = Node::null();
+          break;
+        }
+        unsigned p = path[npath - i - 1];
```

Comment:
```suggestion
        uint32_t p = path[npath - i - 1];
```
Can you also change this in the declaration of `path`?

Reply:
Done